Intercom Server panic on signal shutdown

Killing the server using a signal causes the Intercom server to panic

```
2016/03/09 01:04:26 [INFO]  main: SIGTERM received. Exiting.
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x30 pc=0x4c0abe]

1: running
             panic.go:464  panic(0x9fc100, 0xc820012110)
    intercom server.go:83  ShutdownServer(0, 0)
    main     signals.go:29 stopAllServices()
    main     signals.go:79 handleSIGTERM()
    main     signals.go:53 HandleSignals()
```

**Steps to reproduce this issue:**

`kill <pid>`

--- a/pfsd/intercom/server.go
+++ b/pfsd/intercom/server.go
@@ -12,7 +12,7 @@ import (
 	"time"
 )
 
-var lis net.Listener
+var listener *net.UnixListener
 
 type IntercomServer struct{}
 type EmptyMessage struct{}
@@ -66,6 +66,7 @@ func RunServer(metaDir string) {
 	server := new(IntercomServer)
 	rpc.Register(server)
 	lis, err := net.Listen("unix", socketPath)
+	listener = lis.(*net.UnixListener)
 	if err != nil {
 		Log.Fatalf("Failed to listen on %s: %s\n", socketPath, err)
 	}
@@ -80,7 +81,7 @@ func RunServer(metaDir string) {
 }
 
 func ShutdownServer() error {
-	err := lis.Close()
+	err := listener.Close()
 	if err != nil {
 		return fmt.Errorf("failed to close listener: %s", err)
 	}

